src/screens/IndexScreen.js

```diff
@@ -209,10 +218,26 @@ export class IndexScreen extends React.PureComponent {
           if (!listItems || !listItems.length) return null;
 
           const Component = this.getComponent(query);
+          const fetchMoreData = () =>
+            fetchMore({
+              variables: {
+                offset: listItems.length
+              },
+              updateQuery: (prevResult, { fetchMoreResult }) => {
+                if (!fetchMoreResult || !fetchMoreResult[query].length) return prevResult;
+
+                return { [query]: [...prevResult[query], ...fetchMoreResult[query]] };
+              }
+            });
```

Comment:
please here I need an explanation line by line ,   [Offset-based](https://www.apollographql.com/docs/react/v2.4/features/pagination/#offset-based) didn't help me

Reply:
I expected offset to be an integer..

Reply:
correct, `offset` needs to be an integer/a number.

`fetchMore` is the method from Apollos query component to fetch data in batches. So with pagination we want exactly that. Maybe you can search for general articles about pagination, when the specific one is too detailed?
We call that method everytime a user scrolls at the end of a list to fetch more data to append to the list. The offset increases with every call, because the list gets more items with every call.
`updateQuery` is the callback method from Apollo, which needs to be used to add newly fetched data (`fetchMoreResult `) to the existing list (`prevResult`).
As you can see in Insomnia for example, our returned data from the server is always in a schema like
```
{
  "data": {
    "eventRecords": [
      ...
    ]
  }
}
```

`query` is the type, for example `eventRecords`
So the return of `updateQuery` in `fetchMore` is `{ "eventRecords": [...]  }`, as we need it to work with the data passed to the list. It must be the same structure as the initial call.